### ISSUE TITLE
Un-nested menus

### ISSUE THREAD

**Issue by comradekingu** [falzonv/discreet-launcher#110]
Right now the wrench goes to a list that says:

```
### Discreet Launcher

Favorites

Folders

Hidden apps

Refresh the list of apps

Settings and help
```

Clicking "Settings and help" presents
```
### Customization

"Notification for access to favorites      I/O

(image of mountains) Appearance

(wrench) Operation

(arrow up from line) Export and Import

### Help and general info

(questionmark) Help

(bookmark) Changelog

(some info at the bottom)
```

That means there is a disconnect with the wrench being "menu", then _not_ "Settings and help", but then "Operation" again.
Going two levels deep means clicking ← and going all the way back is worse too.

My proposal:

```
### Discreet Launcher

(star)                           Favorites
(folders)                      Folders
(cut here traced icon) Hidden apps
---
(image of mountains) Appearance         - "Refresh the list of apps…" moved here
(wrench)                     Operation            - "Notification for access to favorites      I/O moved here
(arrow up from line)    Export and Import
---
("i" in a circle)             About                  -  (some info at the bottom) moved here
(bookmark)                 Changelog
(questionmark)           Help
```

**Comment by falzonv** [comment]
Hello,

Thank you for the suggestion, indeed this looks interesting!

So from what we have today, the main steps to move to something like your proposal would be to:
- Rename the "Settings and help" activity to "Discreet Launcher" and make the menu wrench directly open it (removing the intermediary list menu)
- Add the "Favorites" and "Folders" buttons on top of this "Discreet Launcher" activity (in v4.7.0 which should be available soon on F-Droid, "Hidden apps" has already been moved in the current "Settings and help")
- Move the notification setting in the "Operation" section

=> Does it seem correct?

For the "Refresh the list of apps" button it is a bit more complicated. It is still needed for some edge cases but cannot be moved easily in the settings for technical reasons.

Regarding the last entries, I would rather keep the order "Help", "Changelog" and the about info at the end as it is today.

*Remark: I will be away from Internet and GitHub for a couple of weeks and will probably not see your answer before my return.*

Best regards.

**Comment by comradekingu** [comment]
That is it, yes.

I like adapting the current "Discreet Launcher" menu, as it doesn't take over the whole screen, but rather keeps some of the app icons visible. If you do changes, ideally that should be visible.
Don't know if that menu can have icons for menu entries.

Maybe the list of apps could be auto-refreshed when the menu is opened?

**Comment by falzonv** [comment]
Ok thank you for your feedback, I will try to do it like the list menu them.

> Don't know if that menu can have icons for menu entries.

When trying in #82 I was not able to do it, I am not even sure that separators can be added.
Something that I didn't try before for this menu is to create a custom dialog from scratch, similar in behavior to the list menu but allowing to add icons and separators. I will give this a try and we will see.

> Maybe the list of apps could be auto-refreshed when the menu is opened?

Since this is the most resources intensive task of the launcher, I would prefer to avoid that.

The button is there because a manual refresh is needed for some edge cases (like enabling/disabling non-removable apps), but otherwise refreshes are done only on some events (launcher start, app install/uninstall, rename, etc.).
I agree that it is not perfect to have this rarely used entry in the menu, but I am not sure where I can put it for now. Let's already do the other changes described above, and it may give some ideas for this entry as well.

Best regards.

**Comment by falzonv** [comment]
Hello @comradekingu,

Trying to create a completely custom dialog was probably the best option, I was already able to get some good results.
For now I only recreated the current menu with new internal mechanisms (see screenshot below), the next step will be to un-nest the settings and add them to this interface.

![2021-09-05_new_menu_with_icons](https://user-images.githubusercontent.com/4698531/132135535-def82b32-767a-4a78-864e-b8da3109a309.jpg)

---

In parallel, I migrated the Help (as per #105) and it results in the screenshot on the left below, which indeed looks better.
*(Remark: although I was not able to catch it on the screenshot, a scrollbar is visible when the user scrolls)*

However, in the process of doing this, I thought that it could instead be displayed in a popup dialog rather than a full screen activity. The screenshot on the right shows how it would look like and the "Close" button brings back to the menu.
Of course it makes a big popup dialog, but the wallpaper is still visible on edges (and also the "C" from the "Calculator" app) so it might be better for user experience.
*(Remark: same story here with the scrollbar)*

Although I haven't tried it, I guess that the same could be done for the Changelog.

Which one do you prefer?

| Help converted to Activity | Help converted to Dialog |
| --- | --- |
| ![2021-09-05_help_activity](https://user-images.githubusercontent.com/4698531/132135540-0cda3286-80c7-4c74-a8ac-96684ab52e10.jpg) | ![2021-09-05_help_dialog](https://user-images.githubusercontent.com/4698531/132135548-149e35aa-1b8c-4fc6-9db0-ab5c2347251b.jpg) |

**Comment by comradekingu** [comment]
I like the one to the right. It seems less intrusive.
To save space:
← Next to "Help" saves "CLOSE" at the bottom.
Putting titles right next to entries saves one line per entry.
Possibly putting entries inline with entries saves more.

`Navigate` by swiping up and 
`Notifications` show a popup of favo

**Comment by falzonv** [comment]
Ok, let's go for the Dialog option then.

> ← Next to "Help" saves "CLOSE" at the bottom.

I have found another way which is simpler from code perspective and basically the reverse idea: remove the "Help" title and rename the button into "CLOSE HELP".

Here is a screenshot where I added manually the scrollbar as it shows on the device:

![2021-09-05_help_dialog2](https://user-images.githubusercontent.com/4698531/132137775-e6406491-2c67-4319-99ba-cfd4e12df685.jpg)


> Putting titles right next to entries saves one line per entry.
> Possibly putting entries inline with entries saves more.
> 
> `Navigation` is done by swiping up and… 
> `Notifications` show a popup of favo…

Interesting idea but I prefer to keep titles on their own lines (entirely subjective, but I found it more elegant like this).

> The idea of seeing all the help at once without scrolling sends the message that the total complexity of everything isn't that bad.

Perhaps, but I would not make it a goal to see all the help without scrolling. It is good to optimize the help but it should still contain enough information to be useful. Having clear sections takes a bit more space but make the help easier to read in my opinion.

**Comment by comradekingu** [comment]
![bilde](https://user-images.githubusercontent.com/13802408/132138750-28144e54-24da-4e8f-9f08-bb419b83ced6.png)
![bilde](https://user-images.githubusercontent.com/13802408/132138808-588d9a23-4c5e-4412-a8ac-54e44b9e1d90.png)

I think it could be possible, and that the informational density is still rather low.
It all seems somewhat lighter if you can see the icons.

"Nothing left behind, no new forest to enter into" kind of idea.

**Comment by falzonv** [comment]
No, having transparent parts doesn't appear to be possible. I can set an opaque background, but as soon as I add transparency, Android adds a white or dark (depending on the current theme) background behind it to keep the Dialog uniform.

*If I misunderstood the screenshots and you were talking about fitting everything on one screen, you miss the "Folders" and "Search" sections in your screenshots. For a correct display I think at least 1.5 screen is really needed.*

**Comment by falzonv** [comment]
Hello,

I am pleased to inform you that this has been implemented in new release v5.0.0!
This release has been published few minutes ago.

Here are the usual availability delays for each location:
- 4 to 8 days on F-Droid
- 0 to 2 days on Google Play
- Already available as direct APK on GitHub (https://github.com/falzonv/discreet-launcher/releases)

*Remarks:
I will leave this ticket opened until the release is effectively published on F-Droid.
You can use the export/import feature to switch from one version to another.*

Best regards.

**Comment by falzonv** [comment]
Since v5.0.0 is now available on F-Droid, I will close this ticket.
Feel free to add comments if needed or wished.
